covid19-dashboard/dc-covid19/src/DataCalculator.ts

```diff
@@ -1,90 +1,126 @@
 /**
  Copyright 2020 Google LLC
-
  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at
-
  https://www.apache.org/licenses/LICENSE-2.0
-
  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
  */
 
-import {addOrSubtractNDaysToDate, getRealISODatesFromArrayOfDeltaDays} from './Utils'
+import {getRealISODatesFromArrayOfDeltaDays} from './Utils'
+import moment from "moment";
 
 type DataPerGeoIdPerDate = {string: {string: number}} | {}
 
 /**
  * Performs a calculationType iteratively on two dates.
  * @param data: input data in the form of dates->geoIds->value.
  * @param range: the range of dates to calculate the data for in ISO Format. Example: ["2020-01-01", "2020-01-10"]
- * @param deltaDays: Do we wanna compare the data every 2 days? every 7? Any number works.
  * @param calculationType: What type of calculation are we performing?
- * @param geoIdToPopulation: geoId->Population. After calculating the difference, we want to divide by total population
+ * @param deltaDays?: Do we wanna compare the data every 2 days? every 7? Any number works.
+ * @param geoIdToPopulation?: geoId->Population. After calculating the difference, we want to divide by total population
  */
 export default function dataCalculator(data: DataPerGeoIdPerDate,
                                        range: [string, string],
-                                       deltaDays: number,
                                        calculationType: string,
-                                       geoIdToPopulation: {geoId: number}
-                                       ): DataPerGeoIdPerDate {
-    // If there isn't any data, just return an empty {}
-    if (!data) return {}
+                                       deltaDays?: number,
+                                       geoIdToPopulation?: { [p: string]: number }
+): DataPerGeoIdPerDate {
+    // If there isn't any data, return an empty {}
+    if (!Object.keys(data).length) {
+        console.log("No input data.")
+        return {}
+    }
 
-    // If there is data,
-    let outputData: DataPerGeoIdPerDate = {}
+    let date0 = moment(range[0]);
```

Comment:
const date0

Reply:
date0 is updated on iteration.

`while (date0 <= date1){ date0++ }`